features/signup.feature

```diff
@@ -7,45 +7,72 @@ Feature: Visitor Sign Up
 
 
   Scenario: Visitor fills in all fields except Password [Sad Path]
-    When I fill in "Username" with "Scott"
-    And I fill in "Email" with "hotmale@hotmail.com"
-    And I click on the "Create User" button or link
-    Then I should see "Password can't be blank"
+    When I fill in "Email" with "hotmale@hotmail.com"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Password can't be blank"
 
 
   Scenario: Visitor fills in all fields except Password Confirmation [Sad Path]
-    When I fill in "Username" with "Scott"
-    And I fill in "Email" with "hotmale@hotmail.com"
+    When I fill in "Email" with "hotmale@hotmail.com"
     And I fill in "Password" with "whatever"
-    And I click on the "Create User" button or link
-    Then I should see "Password confirmation doesn't match Password"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Password confirmation doesn't match Password"
 
 
   Scenario: Visitor fills in all fields except Email [Sad Path]
-    When I fill in "Username" with "Scott"
-    And I fill in "Password" with "whatever"
-    And I fill in "Password Confirmation" with "whatever"
-    And I click on the "Create User" button or link
-    Then I should see "Email is invalid"
+    When I fill in "Password" with "whatever"
+    And I fill in "Password confirmation" with "whatever"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Email can't be blank"
+
+
+  Scenario: Visitor fills in all fields with invalid email [Sad Path]
+    When I fill in "Email" with "hotmale"
+    When I fill in "Password" with "whatever"
+    And I fill in "Password confirmation" with "whatever"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Email is invalid"
+
+
+  Scenario: Visitor fills in all fields with invalid password [Sad Path]
+    When I fill in "Email" with "hotmale@hotmail.com"
+    When I fill in "Password" with "what"
+    And I fill in "Password confirmation" with "what"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Password is too short (minimum is 6 characters)"
+
+
+  Scenario: Visitor fills in all fields with taken email [Sad Path]
+    When I fill in "Email" with "cutie123@hotmail.com"
+    When I fill in "Password" with "whatever"
+    And I fill in "Password confirmation" with "whatever"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Email has already been taken"
+
+
+  Scenario: Visitor does not fill in any fields [Sad Path]
```

Comment:
The only thing - I should probably divide email and password errors into two scenarios. But maybe that's just a personal preference. :)

Reply:
Not a biggy.. 😉